app/models/vacancy.rb

```diff
@@ -45,7 +45,7 @@ class Vacancy < ApplicationRecord
   # rubocop:disable Metrics/BlockLength
   # rubocop:disable Metrics/LineLength
   # There must be a better way to pass these settings to the block, but everything seems to break
-  algoliasearch index_name: Rails.env.test? ? "Vacancy_test#{ENV.fetch('GITHUB_RUN_ID', '')}" : 'Vacancy', auto_index: Rails.env.production?, auto_remove: Rails.env.production?, synchronous: Rails.env.test?, disable_indexing: !(Rails.env.production? || Rails.env.test?) do
+  algoliasearch if: :listed? do
```

Comment:
Can we retain `disable_indexing` for the staging environment, etc? Otherwise I don't know what is stopping staging data appearing in our index. Perhaps `auto_index` and `auto_remove` are load-bearing too, I'm not sure.

Reply:
I see the next commit handles this. Is it worth having any extra layer of security by using `disable_indexing: Rails.env.staging?` or something?

Reply:
Read the comment in the README about how it handles attempts by non-prod environments attempting to start using the prod index.

Reply:
Also, you seem to have commented on an outdated version of the PR, @david-mears-dfe...